src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.db;
+
+import core.basesyntax.InsufficientStockException;
+import core.basesyntax.ProductIsAbsentException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
```

Comment:
Actually it is good idea to make storage private and create some methods to interact with it, but almost all this methods contains business logic which needs to be defined in strategy classes. I suggest to create only basic static methods here to interact with storage. Also we will avoid injection storage in half of your classes
As an alternative you can read about DAO and implement it in project but this way less preferable.

Reply:
I've already refactored it acoording to Vasyl's comments during Q&A. Hope, it's fine that way :)